cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -852,6 +862,27 @@ public void testGenerateService() {
         TestUtils.checkOwnerReference(clusterIp, KAFKA);
     }
 
+    @ParallelTest
+    public void testGenerateExternalBootstrapServices() {
+        List<Service> services = KC.generateExternalBootstrapServices();
+
+        assertThat(services, hasSize(1));
+        Service service = services.get(0);
+
+        assertThat(service.getSpec().getAllocateLoadBalancerNodePorts(), is(false));
+    }
+
+    @ParallelTest
+    public void testGeneratePerPodServices() {
+        List<Service> services = KC.generatePerPodServices();
+
+        assertThat(services, hasSize(5));
+        assertThat(services.stream()
+                        .filter(s -> !s.getSpec().getAllocateLoadBalancerNodePorts())
+                        .collect(Collectors.toList()),
+                hasSize(5));
+    }
```

Comment:
I think this works. Maybe you can have a single unit test to cover both and you should have the listener configured for this in this method and not in the static resource. Then it would not be needed to include it in all the other tests.

And in one of the generic tests for the services you should also check that the value is not set in other cases.

Reply:
I have moved this test to `KafkaClusterListenerTest` like you suggested. It creates another `KafkaCluster` instance instead of using the static one